kotlin/samples/hello-terminal/terminalworkflow/src/main/java/com/squareup/sample/helloterminal/terminalworkflow/TerminalWorkflowRunner.kt

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.helloterminal.terminalworkflow
+
+import com.googlecode.lanterna.TerminalPosition.TOP_LEFT_CORNER
+import com.googlecode.lanterna.TerminalSize
+import com.googlecode.lanterna.TextColor.ANSI
+import com.googlecode.lanterna.input.KeyStroke
+import com.googlecode.lanterna.screen.Screen
+import com.googlecode.lanterna.screen.Screen.RefreshType.COMPLETE
+import com.googlecode.lanterna.terminal.DefaultTerminalFactory
+import com.squareup.workflow.Workflow
+import com.squareup.workflow.WorkflowAction.Companion.emitOutput
+import com.squareup.workflow.WorkflowAction.Companion.noop
+import com.squareup.workflow.WorkflowHost
+import com.squareup.workflow.WorkflowHost.Update
+import com.squareup.workflow.stateless
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.CONFLATED
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.selects.selectUnbiased
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * TODO write documentation.
+ */
+class TerminalWorkflowRunner(
```

Comment:
Is it worth trying to tease out a `CliWorkflowRunner` or maybe `LanternaWorkflowRunner` that's separate from `TerminalWorkflow` itself? Still as sample code, obv. way premature to try to pull it into a library. 

Reply:
That feels over-engineered/YAGNI, especially for a sample. Most of the code here is very specific to Lanterna's API.